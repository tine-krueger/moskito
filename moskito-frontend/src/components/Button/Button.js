import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Button.propTypes = {
    children: PropTypes.string.isRequired,
    onClick: PropTypes.func
}

export default function Button({children, onClick}) {
    
    return (
        <ButtonStyled onClick={onClick}>{children}</ButtonStyled>
    )
}

const ButtonStyled = styled.button`
    padding: 1em 2em;
    font-family: 'Arvo';
    font-size: 1.1rem;
    border-radius: 10px;
    width: auto;
    border: none;
    background: var(--background);
    box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6), 
                -2px -2px 2px 0px rgba(255,255,255,0.5);
    color: var(--headline);
    text-transform: uppercase;
    margin: 1em auto 3em auto;

    &:hover{
        box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2), 
                inset -3px -4px 6px 0 rgba(255,255,255,0.3);
        outline: none;
    }
`
