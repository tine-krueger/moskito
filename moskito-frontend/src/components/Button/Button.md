```js
<Button children= {'Click Me!'}/>
```