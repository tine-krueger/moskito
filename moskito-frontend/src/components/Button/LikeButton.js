import styled from 'styled-components/macro'
import { AiFillPushpin } from 'react-icons/ai'
import { bookmark } from '../../services/handleBookmarkApi'
import { useState } from 'react'


export default function LikeButton({id}) {
    const [ pinned, setPinned] = useState(false)
    

    return <CampsiteLikeButton pinned={pinned} onClick={handleClick}><AiFillPushpin/></CampsiteLikeButton>

    function handleClick() {
        setPinned(!pinned)
        bookmark(id)
        
    }
}

const CampsiteLikeButton = styled.div`
    position: absolute;
    right: 1em;
    top: 1em;
    height: 3.5em;
    width: 3.5em;
    border-radius: 10px;
    box-shadow: 2px 4px 8px -4px rgba(0, 0, 0, 0.6), 
                -2px -2px 2px 0px rgba(255, 255, 255, 0.5);
    display: flex;
    justify-content: center;
    svg {
        width: 2.1em;
        height: 2.1em;
        stroke: #c97f63;
        stroke-width: 3em;
        fill: ${pops => pops.pinned ? '#c97f63' : '#c2d6d3'};
        align-self: center;
        margin-bottom: 3px;
    }
`