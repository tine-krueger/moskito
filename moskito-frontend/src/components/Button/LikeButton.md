```js
import { AiFillPushpin } from 'react-icons/ai';
const [ isPinned, setIsPinned ] = React.useState(false);

<div style={{width: '100px', position: 'relative', height: '100px'}}>
    <LikeButton isPinned={isPinned} onClick={() => setIsPinned(!isPinned)}>
        <AiFillPushpin/>
    </LikeButton>
</div>
```
