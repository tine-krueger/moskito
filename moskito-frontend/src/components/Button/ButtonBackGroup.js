import styled from 'styled-components/macro'
import Button from "./Button";

export default function ButtonBackGroup({text1, text2, onClick}) {
    return (
        <Buttons>
            <Button >{text1}</Button>
            <Button type="button" onClick={onClick}>{text2}</Button>
        </Buttons>
    )
}

const Buttons = styled.div`
    display: grid;
    justify-items: center;
    Button {
        margin: .5em;
        width: 10em;
    }
`