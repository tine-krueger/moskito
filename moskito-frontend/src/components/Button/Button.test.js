import { render } from '@testing-library/react'
import 'jest-styled-components'
import Button from './Button'

describe('Button', () => {
  it('renders correctly', () => {
    const { container, getByText } = render(<Button>Click Me!</Button>)
    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(/Click Me!/i)).toBeInTheDocument()
  })
})