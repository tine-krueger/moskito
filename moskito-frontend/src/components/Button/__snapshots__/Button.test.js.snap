// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  padding: 1em 2em;
  font-family: 'Arvo';
  font-size: 1.1rem;
  border-radius: 10px;
  width: auto;
  border: none;
  background: #c2d6d3;
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  color: #476c85;
  text-transform: uppercase;
  margin: 1em auto 3em auto;
}

.c0:hover {
  background: linear-gradient(145deg,#A2B2B0,#E2FAF6);
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  outline: none;
}

<button
  class="c0"
>
  Click Me!
</button>
`;
